### PR TITLE
added a setup.sh in the global dir of the project

### LINKED ISSUE #22: Add a sh script to locally install the application using a single command by running that file
As a contributor or user of the URL Shortener application, I often find the manual setup process time-consuming and prone to errors. To streamline the setup process and make it more accessible, I propose adding an automated setup script.

### PR DESCRIPTION
**Pull request by Sahilll15**
closes #22 

changes made added a Setup.sh file in the main dir of the project
- run the file in macOS or Linux
- chmod +x ./Setup.sh 
- then ./Setup.sh 
  this will setup the application and add the local mongo URL to the .env file 
  and start the frontend 
- For macOS/Linux: you can start the server by moving into the server folder 
- and just doing `npm start`
+ For Windows:
+ - Open Command Prompt (CMD) or PowerShell
+ - Navigate to the main directory of the project where Setup.sh is located
+ - Run the following commands:
+     - `.\Setup.sh`
+     - This will setup the application and add the local MongoDB URL to the .env file and start the frontend
+ - To start the server, navigate to the server folder in Command Prompt or PowerShell
+ - Run the command: `npm start`

### CODE REVIEW

File: Setup.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+cd server
+npm install
+cp .env.example .env 
+
+echo "NODE_ENV=development" > .env
+echo "CLIENT_PROD_URL=http://localhost:3000" >> .env
+echo "DB_URL=mongodb://localhost:27017" >> .env
+echo "REDIRECT_URL=http://localhost:4000" >> .env
+echo "Port=4000" >>.env
```
Comment (abhijeetnishal) [inline Setup.sh:9]: Here also change Port to 8080

File: Setup.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+cd ..
+
+cd client
+npm install
+echo "REACT_APP_SERVER=http://localhost:4000" > .env
```
Comment (abhijeetnishal) [inline Setup.sh:15]: Here also port is 8080

**Comment by abhijeetnishal** [comment]
Hey @Sahilll15, I have added some changes for this PR. 
Try to resolve this so that this PR is merged.

**Comment by Sahilll15** [comment]
@abhijeetnishal have a look!!

**Comment by abhijeetnishal** [comment]
> @abhijeetnishal have a look!!
Hey @Sahilll15, I am merging this PR now. You can also update in readme about the script.

**Comment by Sahilll15** [comment]
> > @abhijeetnishal have a look!!
> > Hey @Sahilll15, I am merging this PR now. You can also update in readme about the script.

okay

**Pull request merged**